aptos-move/move-examples/fungible_asset/sources/coin_example.move

```diff
@@ -0,0 +1,136 @@
+module aptos_framework::coin_example {
+    use aptos_framework::object;
+    use std::string::String;
+    use aptos_framework::object::{generate_extend_ref, Object, disable_ungated_transfer, generate_transfer_ref};
+    use fungible_asset::managed_fungible_asset::{initialize_managing_capabilities, mint_by_asset_owner, freeze_by_asset_owner, burn_by_asset_owner};
+    use std::signer::address_of;
+    use fungible_asset::fungible_asset;
+    use std::error;
+    use aptos_std::smart_table;
+    use aptos_std::smart_table::SmartTable;
+    #[test_only]
+    use std::string;
+    #[test_only]
+    use aptos_framework::account::create_account_for_test;
+
+    struct Coin has key {
+        name: String,
+        /// Symbol of the coin, usually a shorter version of the name.
+        /// For example, Singapore Dollar is SGD.
+        symbol: String,
+        /// Number of decimals used to get its user representation.
+        /// For example, if `decimals` equals `2`, a balance of `505` coins should
+        /// be displayed to a user as `5.05` (`505 / 10 ** 2`).
+        decimals: u8
+    }
+
+    struct CoinIndexer has key {
+        index: smart_table::SmartTable<String, Object<Coin>>
+    }
```

Comment:
should be part of fungible and the left side should probably be an address :grimacing:

Reply:
I think fungible asset map by address of fungible source is sufficient.